Upstream: yes
Reason: There are KF5 based applications that try to install to
        KXMLGUI5DIR and thus fail if isn't defined

From 032b4c4baaaff5fde1243864149d33f545767b5c Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Mon, 7 Jul 2014 00:09:53 +0200
Subject: [PATCH] Define convenience variable for DATAROOTDIR/kxmlgui5.

See RR 119142 for more details.
---
 kde-modules/KDEInstallDirs.cmake                     | 6 ++++++
 tests/KDEInstallDirsTest/vars_defined/CMakeLists.txt | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/kde-modules/KDEInstallDirs.cmake b/kde-modules/KDEInstallDirs.cmake
index 061e542..90ed5b8 100644
--- a/kde-modules/KDEInstallDirs.cmake
+++ b/kde-modules/KDEInstallDirs.cmake
@@ -69,6 +69,9 @@
 # ``KSERVICETYPES5DIR``
 #     service types for KDE Frameworks 5 (``DATAROOTDIR/kservicetypes5``)
 #     [``SERVICETYPES_INSTALL_DIR``]
+# ``KXMLGUI5DIR``
+#     knotify description files (``DATAROOTDIR/kxmlgui5``)
+#     [``KXMLGUI_INSTALL_DIR``]
 # ``KNOTIFY5RCDIR``
 #     knotify description files (``DATAROOTDIR/knotifications5``)
 #     [``KNOTIFYRC_INSTALL_DIR``]
@@ -384,6 +387,9 @@ _define_relative(KSERVICETYPES5DIR DATAROOTDIR "kservicetypes5"
 _define_relative(KNOTIFY5RCDIR DATAROOTDIR "knotifications5"
     "knotify description files"
     KNOTIFYRC_INSTALL_DIR)
+_define_relative(KXMLGUI5DIR DATAROOTDIR "kxmlgui5"
+    "kxmlgui .rc files"
+    KXMLGUI_INSTALL_DIR)
 
 # Cross-desktop or other system things
 _define_relative(ICONDIR DATAROOTDIR "icons"
diff --git a/tests/KDEInstallDirsTest/vars_defined/CMakeLists.txt b/tests/KDEInstallDirsTest/vars_defined/CMakeLists.txt
index f1617da..3940b81 100644
--- a/tests/KDEInstallDirsTest/vars_defined/CMakeLists.txt
+++ b/tests/KDEInstallDirsTest/vars_defined/CMakeLists.txt
@@ -31,6 +31,7 @@ set(relative_path_vars
     KF5_INCLUDE_INSTALL_DIR
     KF5_LIBEXEC_INSTALL_DIR
     KNOTIFYRC_INSTALL_DIR
+    KXMLGUI_INSTALL_DIR
     LIBEXEC_INSTALL_DIR
     LIB_INSTALL_DIR
     LOCALE_INSTALL_DIR
@@ -73,6 +74,7 @@ set(relative_path_vars
     CMAKE_INSTALL_KCFGDIR
     CMAKE_INSTALL_KCONFUPDATEDIR
     CMAKE_INSTALL_KNOTIFY5RCDIR
+    CMAKE_INSTALL_KXMLGUI5DIR
     CMAKE_INSTALL_KSERVICES5DIR
     CMAKE_INSTALL_KSERVICETYPES5DIR
     CMAKE_INSTALL_LIBDIR
@@ -117,6 +119,7 @@ set(absolute_path_vars
     CMAKE_INSTALL_FULL_KCFGDIR
     CMAKE_INSTALL_FULL_KCONFUPDATEDIR
     CMAKE_INSTALL_FULL_KNOTIFY5RCDIR
+    CMAKE_INSTALL_FULL_KXMLGUI5DIR
     CMAKE_INSTALL_FULL_KSERVICES5DIR
     CMAKE_INSTALL_FULL_KSERVICETYPES5DIR
     CMAKE_INSTALL_FULL_LIBDIR
-- 
2.0.2

